### PR TITLE
UIU-1907: Set expiration date on offset

### PR DESCRIPTION
**Pull request by elsenhans**
https://issues.folio.org/browse/UIU-1907

### Background:
In Settings, a user can define a default offset in days for the expiration date of each patron group. this is optional for each patron group.
If the patron group does have a defined offset, the expiration date field can be filled with the current date plus the given offset.
This automatically entered value can be edited or deleted by the user: it is only a help for choosing the right date, no restriction.

### Scenarios:
Users App, Create/Edit Screen.

(1)
Given: No patron group is selected.
Then: No button is displayed.

(2)
Given: A patron group is already selected (In edit screen)
(2a)
Given: The patron group has no offset defined in settings.
Then: No button is displayed.
(2b)
Given: The patron group does have an offset defined in settings.
Then: The button "Recalculate" is displayed underneath the field "expiration date".
When: The user clicks the button
Then: The expiration date is changed to the current date (server time) plus the number of days defined in the settings as offset.

(3)
Given: The user selects a patron group in the dropdown.
(3a)
Given: The patron group has no offset defined in settings.
Then: No button is displayed.
(3b)
Given: The patron group does have an offset defined in settings.
Then: A modal appears:
"Users with patron group group usually expire after offset days.
Do you want to change the expiration date accordingly?"
Options: "Cancel" - "Recalculate"
When: The user clicks the button "Cancel", Then: The field Expiration date remains unchanged.
When: The user clicks the button "Recalculate". Then: The expiration date is changed to the current date (server time) plus the number of days defined in the settings as offset.
After closing the modal, see (2b): The button still displays, because the user could manually change the filled in date and might want to recalculate again.

### In addition:

Given: An expiration date that is already entered an lies in the past, the status is "inactive", the information text "User has expired" is displayed in red underneath the field status.
When: The expiration date field is changed (manually or via button "Recalculate") to a value >= the current date
Then: An information text apperas instead of or underneath the "User has expired" info, reading: "User will reactivate after saving"

### CODE REVIEW

File: src/components/EditSections/EditUserInfo/EditUserInfo.js [hunk 1]
```diff
@@ -71,112 +142,166 @@ class EditUserInfo extends React.Component {
       }
     ];
 
+    const offset = this.getPatronGroupOffset();
+    const group = _.get(this.props.patronGroups.find(i => i.id === this.props.selectedPatronGroup), 'group', '');
+    const footer = (
```
Comment (rchr) [inline src/components/EditSections/EditUserInfo/EditUserInfo.js:147]: Please rename this constant to e.g. `modalFooter`. Then it is clearer that this is the footer of the modal.

File: src/views/UserEdit/UserForm.js [hunk 2]
```diff
@@ -503,6 +503,7 @@ class UserForm extends React.Component {
                   onToggle={this.handleSectionToggle}
                   initialValues={initialValues}
                   patronGroups={formData.patronGroups}
+                  stripes={this.props.stripes}
```
Comment (rchr) [inline src/views/UserEdit/UserForm.js:506]: Just a matter of style. I would prefer to add 'stripes' to the object desctructuring in L445 - L452. Then you can write `stripes={stripes}` here.

**Review by rchr — CHANGES_REQUESTED** [review]
Good work! Thank you!

However, I do have a few small remarks. Mostly questions of style.

**Comment by elsenhans** [comment]
@rchr, I fixed your style remarks. Thanks a lot for your feedback!

**Review by rchr — APPROVED** [review]
LGTM!

**Review by zburke — APPROVED** [review]
Nicely done!

**Pull request merged**